pimd/pim_tlv.c

```diff
@@ -491,10 +509,24 @@ pim_parse_addr_ucast (struct prefix *p,
 
     p->family = AF_INET; /* notice: AF_INET != PIM_MSG_ADDRESS_FAMILY_IPV4 */
     memcpy(&p->u.prefix4, addr, sizeof(struct in_addr));
-
+    p->prefixlen = IPV4_MAX_PREFIXLEN;
     addr += sizeof(struct in_addr);
 
     break;
+  case PIM_MSG_ADDRESS_FAMILY_IPV6:
+    if ((addr + sizeof(struct in6_addr)) > pastend) {
+      zlog_warn ("%s: IPv6 unicast address overflow: left=%zd needed %zu",
+                 __PRETTY_FUNCTION__,
+                 pastend - addr, sizeof(struct in6_addr));
+      return -3;
+    }
+
+    p->family = AF_INET6;
+    p->prefixlen = IPV6_MAX_PREFIXLEN;
+    memcpy(&p->u.prefix6, addr, 16);
```

Comment:
`sizeof(struct in6_addr)`? (`16` in one line + `sizeof` in the next looks really weird)